replication-api-impl/src/test/java/org/codice/ditto/replication/api/impl/persistence/ReplicatorHistoryManagerImplTest.java

```diff
@@ -0,0 +1,196 @@
+/**
+ * Copyright (c) Connexta
+ *
+ * <p>This is free software: you can redistribute it and/or modify it under the terms of the GNU
+ * Lesser General Public License as published by the Free Software Foundation, either version 3 of
+ * the License, or any later version.
+ *
+ * <p>This program is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY;
+ * without even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU Lesser General Public License for more details. A copy of the GNU Lesser General Public
+ * License is distributed along with this program and can be found at
+ * <http://www.gnu.org/licenses/lgpl.html>.
+ */
+package org.codice.ditto.replication.api.impl.persistence;
+
+import static org.hamcrest.Matchers.*;
+import static org.junit.Assert.assertThat;
+import static org.mockito.Matchers.eq;
+import static org.mockito.Mockito.*;
+
+import java.util.Date;
+import java.util.stream.Stream;
+import javax.ws.rs.NotFoundException;
+import org.codice.ditto.replication.api.Status;
+import org.codice.ditto.replication.api.data.ReplicationStatus;
+import org.codice.ditto.replication.api.impl.data.ReplicationStatusImpl;
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.mockito.Mock;
+import org.mockito.runners.MockitoJUnitRunner;
+
+@RunWith(MockitoJUnitRunner.class)
+public class ReplicatorHistoryManagerImplTest {
+
+  ReplicatorHistoryManagerImpl history;
+
+  @Mock ReplicationPersistentStore persistentStore;
+
+  @Before
+  public void setUp() throws Exception {
+    history = new ReplicatorHistoryManagerImpl(persistentStore);
+  }
+
+  @Test
+  public void create() {
+    assertThat(history.create(), instanceOf(ReplicationStatusImpl.class));
+  }
+
+  @Test
+  public void get() {
+    history.get("test");
+    verify(persistentStore).get(ReplicationStatusImpl.class, "test");
+  }
+
+  @Test
+  public void objects() {
+    when(persistentStore.objects(eq(ReplicationStatusImpl.class))).thenReturn(Stream.empty());
+    history.objects();
+    verify(persistentStore).objects(ReplicationStatusImpl.class);
+  }
+
+  @Test
+  public void getByReplicatorId() {
+    ReplicationStatusImpl status = new ReplicationStatusImpl();
+    status.setReplicatorId("test");
+    when(persistentStore.objects(eq(ReplicationStatusImpl.class))).thenReturn(Stream.of(status));
+    assertThat(history.getByReplicatorId("test"), is(status));
+  }
+
+  @Test(expected = NotFoundException.class)
+  public void getByReplicatorIdNotFoundException() {
+    ReplicationStatusImpl status = new ReplicationStatusImpl();
+    status.setReplicatorId("test");
+    when(persistentStore.objects(eq(ReplicationStatusImpl.class))).thenReturn(Stream.empty());
+    history.getByReplicatorId("test");
+  }
+
+  @Test
+  public void save() {
```

Comment:
❓ Is this test needed since we have the addStatus test? `addStatus` can be private and tested indirectly through the save method. I think most the tests below should be going through the save method since it will ensure that `save` is doing what we expect.

Reply:
the addStats method is now on the ReplicationStatus class and is tested seperately.